### PR TITLE
Update to Swift 1.2

### PR DESCRIPTION
**Pull request by kimdv**
Updated and I can compile to Swift 1.2 Xcode 6.3

### CODE REVIEW

**Comment by MarvinSchramm** [comment]
+1

**Comment by nunogoncalves** [comment]
+1

**Comment by wakinchan** [comment]
:+1:

**Comment by evenh** [comment]
+1

**Comment by jlalvarez18** [comment]
:+1:

**Comment by minhoryang** [comment]
+1

**Comment by Kevsterstyle** [comment]
+1

**Comment by ccerinojr** [comment]
👍

**Comment by dwa012** [comment]
+1

**Comment by jlalvarez18** [comment]
:+1:

**Comment by chrux** [comment]
Awesome! but can we also update the tests? when running with carthage to build it, it's giving me
```
ios/Carthage/Checkouts/SwiftyJSON/Tests/RawRepresentableTests.swift:108:44: error: 'AnyObject' is not convertible to 'NSDictionary'; did you mean to use 'as!' to force downcast?
        XCTAssertTrue(dictionary == object as NSDictionary)
                                    ~~~~~~~^~~~~~~~~~~~~~~
                                           as!
```

**Comment by al7** [comment]
+1

**Comment by wangyhzzz** [comment]
+1

File: Source/SwiftyJSON.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ public enum Type :Int{
 // MARK: - JSON Base
 
 public struct JSON {
-
+
```
Comment (minhoryang) [inline Source/SwiftyJSON.swift]: Is it necessary? I don't think so.

**Comment by macu** [comment]
Please maintain the original whitespace so that meaningful changes stand out in diff.

**Comment by chrux** [comment]
@macu +1

**Comment by parkej60** [comment]
+1

**Comment by whusted** [comment]
@mb89 Thanks for noting this

**Comment by tangplin** [comment]
Done db2b0274684779ee0d6e0f169f89f499883260a0

**Pull request closed without merging**